package/MDAnalysis/analysis/gnm.py

```diff
@@ -314,14 +327,17 @@ def __init__(self, universe, selection='protein', cutoff=4.5, ReportVector=None,
         self.MassWeight = MassWeight
 
     def generate_kirchoff(self):
-        natoms = len(self.ca.atoms)
-        nresidues = len(self.ca.residues)
+        natoms = self.ca.n_atoms
+        nresidues = self.ca.n_residues
         positions = self.ca.positions
         res_positions, grid, low_x, low_y, low_z = generate_grid(positions, self.cutoff)
         residue_index_map = [resnum for [resnum, residue] in enumerate(self.ca.residues) for atom in residue.atoms]
-        matrix = np.zeros((nresidues, nresidues), "float")
+        matrix = np.zeros((nresidues, nresidues), dtype=np.float_)
         cutoffsq = self.cutoff ** 2
 
+        # cache sqrt of residue sizes (slow) so that sr[i]*sr[j] == sqrt(r[i]*r[j])
+        sqrt_res_sizes = np.sqrt([r.atoms.n_atoms for r in  self.ca.residues]) if self.MassWeight else None
```

Comment:
Btw is that really a weights defined by the mass? It looks only like a estimate of the actual mass. Would `size` be a better name?

Since you are refactoring this code already. It be nice if this variable was renamed `weights` and then have options `None` or `'size'`.